stories/notifications/Notifications.stories.js

```diff
@@ -0,0 +1,1162 @@
+                text: "Team join request denied",
+            },
+            seen: true,
+            deleted: false,
+        },
+        {
+            type: "team",
+            user: "ipcdev",
+            subject: "Team join request denied",
+            email: true,
+            email_template: "team_join_denial",
+            payload: {
+                email_address: "core-sw@iplantcollaborative.org",
+                team_name: "sriram:sriramtest",
+                admin_message: "you are not cool..",
+            },
+            message: {
+                id: "094DEB26-4DEF-4D85-98D1-033A7F137F74",
+                timestamp: "1532454413679",
+                text: "Team join request denied",
+            },
+            seen: true,
+            deleted: false,
+        },
+        {
+            type: "team",
+            user: "ipcdev",
+            subject: "Added to team",
+            email: true,
+            email_template: "added_to_team",
+            payload: {
+                email_address: "core-sw@iplantcollaborative.org",
+                team_name: "sriram:test123",
+            },
+            message: {
+                id: "7CF916A4-56AF-43E3-8151-8DD6A92F276B",
+                timestamp: "1532454009332",
+                text: "Added to team",
+            },
+            seen: true,
+            deleted: false,
+        },
+        {
+            type: "data",
+            user: "ipcdev",
+            subject: "julianp has shared 1 file(s)/folder(s) with you.",
+            email: false,
+            email_template: null,
+            payload: {
+                action: "share",
+                paths: ["/iplant/home/julianp/atmosphere_user_videos"],
+            },
+            message: {
+                id: "F74B592F-AA8D-4C7B-8458-3AF1D34ED4A2",
+                timestamp: "1528764376984",
+                text:
+                    "julianp has shared the following file(s)/folder(s) with you: atmosphere_user_videos",
+            },
+            seen: true,
+            deleted: false,
+        },
+    ],
+    "system-messages": [],
+};
+//
+// const presenter = {
+//     getNotifications: (
+//         limit,
+//         offset,
+//         filter,
+//         sortDir,
+//         resultCallback,
+//         errorCallback
+//     ) => {
+//         resultCallback(notifications, 10);
+//     },
+//     deleteNotifications: (
+//         selected,
+//         rowsPerPage,
+//         resultCallback,
+//         errorCallback
+//     ) => {
+//         resultCallback(trimmed_notifications, 8);
+//     },
+//     onNotificationToolbarMarkAsSeenClicked: () =>
+//         logger("Mark As Seen"),
+//     onMessageClicked: () => logger("MessageClicked!"),
+// };
+// const logger =
+//     this.props.logger ||
+//     ((Notification) => {
+//         console.log(Notification);
+//     });
+
+return (
+    <NotificationsPreview
+        // presenter={presenter}
+        baseDebugId="notificationWindow"
```

Comment:
Looks like you have all that notification mock data, but you're not using it. It's possible you didn't have any data to work with until after you had already begun putting stuff together. I would really recommend that your dev process begins with using mock data and iterating your components over that data.  This will save you a TON of time in the long-run.

Reply:
Yes, I had this added recently. I did not think there would be too much of a difference with replacing hard coded data with mock data in the end. In what ways does it help to save time?

Reply:
Just to be clear, hard-coded data is not the issue. Mock data is also hard-coded data after all :smile: The issue is how you're using that data. From a high level, it's causing you to add an extra step.

Right now, you're baking the data directly into the component, making it a static component, in order to get a feel for the look of things and get feedback. Once that's complete, then you'll need to remove the baked in data and also refactor your static component into a dynamic component that can work with real/mocked data.  It's like you're creating the same component twice. Storybook is really great for eliminating that extra step because it allows you to bake your data into a story instead so your component shouldn't have to be refactored from static to dynamic.

Beyond that, knowing from the start what the data looks like and working with it directly can save time. Maybe you've crafted your component around certain assumptions about the data and then come to find out the data is different than you expected. For example, the conversation about notification length came up in Slack. Some notifications are longer than maybe you originally thought, so then we started the conversation about using ellipses or not. Just as an example.